### ISSUE TITLE
Provide preference for default space directory

### ISSUE THREAD

**Issue by alfred-landrum** [brimsec/brim#676]
With https://github.com/brimsec/brim/issues/673 , we're going to default to creating space directories in `<userData>/data/spaces`. We can add a Brim preference that will allow a user to specify the default data directory location for space creation. After this preference is set, any spaces created should use a `data_dir` parameter that includes the specified location.

**Comment by philrz** [comment]
Details on use case: The default "data dir" might be on a main storage device that has limited space available. However, a user might have some bulk storage like an external multi-TB hard drive where they're happy to store excess amounts of bulky data.

**Comment by jameskerr** [comment]
The `data_dir` for a space was the actual .brim folder where all.zng was written to. It seems like this request is for a "spaces root" folder rather than a global `data_dir`. Am I understanding this correctly?

**Comment by jameskerr** [comment]
Or is the app still responsible for providing the last bit of the data_dir as well? For example:

`<user-provided-data-dir>/sample.pcap.brim`
`<user-provided-data-dir>/my.logs.brim`

**Comment by philrz** [comment]
Verified in Brim commit `dc488b1` talking to `zqd` commit `a14daef`.

Per the attached video, the picker is helpful in ensuring I only pick a directory and not a file, and that the directory actually exists. Once I've imported data, I can see the relevant directories and files are created in the destination directory I selected.

[Verify.zip](https://github.com/brimsec/brim/files/4682679/Verify.zip)

Thanks @mason-fish!
